starlette/datastructures.py

```diff
@@ -114,11 +114,18 @@ def replace(self, **kwargs: typing.Any) -> "URL":
             or "hostname" in kwargs
             or "port" in kwargs
         ):
-            hostname = kwargs.pop("hostname", self.hostname)
+            hostname = kwargs.pop("hostname", None)
             port = kwargs.pop("port", self.port)
             username = kwargs.pop("username", self.username)
             password = kwargs.pop("password", self.password)
 
+            if not hostname:
```

Comment:
Should we check against `None` instead ("not given in `kwargs`")?

```suggestion
            if hostname is None:
```